node/src/components/consensus/protocols/highway.rs

```diff
@@ -209,14 +209,19 @@ impl<I: NodeIdT, C: Context> HighwayProtocol<I, C> {
                 } else {
                     match self.highway.validate_vertex(pvv) {
                         Ok(vv) => {
-                            if let Some(value) = vv.inner().value().cloned() {
+                            let vertex = vv.inner();
+                            if let (Some(value), Some(timestamp)) =
+                                (vertex.value().cloned(), vertex.timestamp())
+                            {
                                 // It's a block: Request validation before adding it to the state.
                                 self.pending_values
                                     .entry(value.clone())
                                     .or_default()
                                     .push(vv);
                                 results.push(ConsensusProtocolResult::ValidateConsensusValue(
-                                    sender, value,
+                                    sender,
+                                    value.clone(),
```

Comment:
Is that additional `clone()` actually needed? (Not your fault, I know!)

Reply:
Nope, thanks.